runtime/parachains/src/paras_inherent.rs

```diff
@@ -254,13 +256,24 @@ pub mod pallet {
 
 			// Process backed candidates according to scheduled cores.
 			let parent_storage_root = parent_header.state_root().clone();
-			let occupied = <inclusion::Pallet<T>>::process_candidates(
+			let inclusion::ProcessedCandidates::<<T::Header as HeaderT>::Hash> {
+				core_indices: occupied,
+				candidate_receipt_with_backing_validator_indices,
+			} = <inclusion::Pallet<T>>::process_candidates(
 				parent_storage_root,
 				backed_candidates,
 				<scheduler::Pallet<T>>::scheduled(),
 				<scheduler::Pallet<T>>::group_validators,
 			)?;
 
+			// The number of disputes included in a block is
+			// limited by the weight as well as the number of candidate blocks.
+			OnChainVotes::<T>::put(ScrapedOnChainVotes::<<T::Header as HeaderT>::Hash> {
+				session: current_session,
+				backing_validators: candidate_receipt_with_backing_validator_indices,
+				disputes,
+			});
```

Comment:
I am a bit confused that we need to store this - shouldn't the required information be already available on chain?

Reply:
New to runtime programming, so maybe a stupid question: Assuming we need to store this, should we also take care of freeing that storage once we are done with it?

Reply:
> New to runtime programming, so maybe a stupid question: Assuming we need to store this, should we also take care of freeing that storage once we are done with it?

Bernhard already pointed out - that's a single entry that gets overwritten every time - so short answer: No.

Reply:
We need the candidate receipts for the backing votes somehow, now if we have to query around anyways, there is little reason to not just utilize the `Disputes` storage. TBD